backend/mem/fileSystem.go

```diff
@@ -0,0 +1,176 @@
+package mem
+
+import (
+	"path"
+	"time"
+
+	"github.com/c2fo/vfs/v5"
+	"github.com/c2fo/vfs/v5/backend"
+	"github.com/c2fo/vfs/v5/utils"
+)
+
+//Scheme defines the filesystem type.
+const Scheme = "mem"
+const name = "In-Memory Filesystem"
+
+// FileSystem implements vfs.Filesystem for the mem filesystem.
+type obj struct {
+	isFile bool
+	i      interface{}
+}
+type objMap map[string]*obj
+type FileSystem struct {
+	fsMap map[string]objMap
+}
+
+// FileSystem will return a retrier provided via options, or a no-op if none is provided.
```

Comment:
Start this comment with "Retry will..." not "FileSystem will..." The comment should start with the name of the function being described.